atlp-rwanda/e-commerce-mavericcks-bn
# [Finishes #187354247] login into the app via Google

## Purpose

The purpose of this pull request is to implement user authentication via Google OAuth 2.0 using Passport.js in our application.

## Changes Made

- Implemented Google OAuth 2.0 authentication strategy using Passport.js.
- Added necessary routes and middleware to handle authentication with Google.
- Integrated JWT token generation upon successful authentication.
- Updated user authentication logic to include Google authentication.

## Testing Instructions

To test the changes:

1. Make sure the environment variables for Google OAuth 2.0 credentials are correctly set up.
2. Run the application and navigate to the authentication endpoint **(Make sure you sure PORT  8080 for Localhost ).**
3. Click on the "Login with Google" button and follow the authentication flow.
4. Verify that the user is successfully authenticated and redirected to the intended page.
5. Check if the JWT token is generated and sent back to the client upon successful authentication.
6. Ensure that users can access protected routes only when authenticated.

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:
